### PR TITLE
Rabbit pub consume

### PR DESCRIPTION
**Pull request by philipwhitt**

### CODE REVIEW

**Comment by philipwhitt** [comment]
I also made the bin/chinchilla app not exit on "yaml not found"

File: Fliglio/Chinchilla/WorkerPublisher.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	public function publish(MappableApi $api, $headers=[]) {
+		$apiClassName = get_class($api);
+
+		$headers['created'] = ['T', time()];
+		$headers['test']    = ['I', (int) false];
+
+		$vo = $apiClassName::getApiMapper()->marshal($api);
+
+		$msg = new AMQPMessage(json_encode($vo), [
+			'content_type'        => 'text/plain',
```
Comment (benschw) [inline Fliglio/Chinchilla/WorkerPublisher.php]: application/json?

File: Fliglio/Chinchilla/WorkerPublisher.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		$vo = $apiClassName::getApiMapper()->marshal($api);
+
+		$msg = new AMQPMessage(json_encode($vo), [
+			'content_type'        => 'text/plain', 
+			'message_id'          => uniqid(),
+			'delivery_mode'       => 2, // persistent : 2, non-persistent : 1
+			'application_headers' => $headers
+		]);
+
+		$this->channel->basic_publish($msg, '', $this->queueName);
```
Comment (benschw) [inline Fliglio/Chinchilla/WorkerPublisher.php]: for clarity, maybe set a class constant "DEFAULT_EXCHANGE = ''"

File: Fliglio/Chinchilla/TopicPublisher.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+use PhpAmqpLib\Message\AMQPMessage;
+use Fliglio\Web\MappableApi;
+
+class TopicPublisher {
+
+	public $channel;
+	public $exchangeName;
+	public $routingKey;
+
+	public function __construct(AMQPChannel $channel, $exchangeName, $routingKey, $passive=false, $durable=true, $auto_delete=false) {
```
Comment (benschw) [inline Fliglio/Chinchilla/TopicPublisher.php]: should routing key be here? or part of the "publish" signature

File: Fliglio/Chinchilla/Config.php [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+		return $this->user;
+	}
+	public function getPassword() {
+		return $this->password;
+	}
+	public function getPort() {
+		return $this->port;
+	}
+
+	public function setName($name) {
```
Comment (benschw) [inline Fliglio/Chinchilla/Config.php]: ?

Comment (benschw) [inline Fliglio/Chinchilla/Config.php]: what is this?

File: Fliglio/Chinchilla/Connection.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+	public static function hasConnection() {
+		return count(self::$connections) > 0;
+	}
+
+	public static function hasConfigs() {
+		return count(self::$configs) > 0;
+	}
+
+	public static function get($name = null) {
```
Comment (benschw) [inline Fliglio/Chinchilla/Connection.php]: how pissed would you be if we didn't handle this as a registry/factory, but instead injected it from the application configs into the clients / resources?

(not sure we should, but we should discuss it)

File: Fliglio/Chinchilla/WorkerPublisher.php [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+			$arguments=null) {
+
+		parent::__construct($connection);
+
+		$this->queueName = $queueName;
+
+		$this->channel->queue_declare(
+			$queueName, 
+			$passive,
+        	$durable,
```
Comment (benschw) [inline Fliglio/Chinchilla/WorkerPublisher.php]: ewww don't you even tab?

**Comment by benschw** [comment]
:poop:

**Pull request merged**